internal/model/apmevent.go

```diff
@@ -72,6 +72,14 @@ type APMEvent struct {
 	// Supports slice values.
 	NumericLabels NumericLabels
 
+	// GlobalLabels are only populated when the batch metadata contains labels
+	// and the decoded event is a transaction event. This is done to be able to
+	// differentiate between event and global labels since we're generating
+	// transaction metrics which include global labels as dimensions to power
+	// certain parts of the APM UI. Not indexed.
+	GlobalLabels        Labels
+	GlobalNumericLabels NumericLabels
```

Comment:
I was thinking that instead of introducing `Global*` counterparts, we could add a `Global bool` field on the `model.LabelValue` and `model.NumericLabelValue` types. Then when merging global/non-global labels, we would set the Global flag to false if the event-specific label's value is not the same as the global label's value. WDYT?

Reply:
that is much cleaner than what I did, thank you for the suggestion!